### PR TITLE
CORCI-594 Start adding regression tests

### PR DESCRIPTION
**Pull request by brianjmurrell**
We should have some tests to ensure that changes to the code review
code doesn't cause regressions.

This starts to add some basic tests.

Signed-off-by: Brian J. Murrell <brian.murrell@intel.com>

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+// To use a test branch (i.e. PR) until it lands to master
+// I.e. for testing library changes
+//@Library(value="pipeline-lib@your_branch") _
+def sanitized_JOB_NAME = JOB_NAME.replaceAll('/', '-').toLowerCase()
+
+pipeline {
+    agent { label 'lightweight' }
+
+    environment {
+        GITHUB_USER = credentials('daos-jenkins-review-posting')
```
Comment (JohnMalmberg) [inline Jenkinsfile]: We should not have global token based credentials.
A "with credentials(credential_id) { block }" should be used for all credential access.

Comment (brianjmurrell) [reply Jenkinsfile]: Done in the subsequent patch.

**Comment by brianjmurrell** [comment]
It's trying to apply some regression testing to the automated code review process so that we know if/when a change causes a problem.

Funny enough I was just looking into making the commit message better before seeing your comment but I'm not sure that I can just change the commit message without a force push.

**Comment by brianjmurrell** [comment]
@JohnMalmberg, @jolivier23 My apologies.  I broke the rule about not (rebasing and) force pushing after reviews have started, but when I tried to change the base on this patch back to master it had conflicts because it did not drop from this PR the patch that landed to master, that it was based on.

I think some experimentation is going to be needed to fully understand the workflow of stacking patches.

**Pull request merged**